### ISSUE TITLE
Flake: SingleBalancerTest.BackendsRestart GRPC_POLL_STRATEGY=epollex 

### ISSUE THREAD

**Issue by ncteisen** [grpc/grpc#13299]
[Kokoro](https://sponge.corp.google.com/target?id=ef801e25-a3cc-4cb3-99b4-7e757fd88c11&target=github/grpc/c%2B%2B_linux_asan_native&searchFor=&show=ALL&sortBy=STATUS)

```
Failure
D1102 01:50:21.963275396    8772 ev_posix.cc:135]            Using polling engine: epollex
D1102 01:50:21.963454907    8772 dns_resolver.cc:306]        Using native dns resolver
D1102 01:50:21.964308639    8772 test_config.c:388]          test slowdown factor: sanitizer=3, fixture=1, poller=1, total=3
Note: Google Test filter = SingleBalancerTest.BackendsRestart
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from SingleBalancerTest
[ RUN      ] SingleBalancerTest.BackendsRestart
I1102 01:50:21.967618025    8772 grpclb_end2end_test.cc:529] starting backend server on port 25482
I1102 01:50:21.969029649    8924 server_builder.cc:269]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I1102 01:50:21.973931789    8772 grpclb_end2end_test.cc:534] backend server startup complete
I1102 01:50:21.975497567    8772 grpclb_end2end_test.cc:529] starting backend server on port 14407
I1102 01:50:21.975818411    8977 server_builder.cc:269]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I1102 01:50:21.980326506    8772 grpclb_end2end_test.cc:534] backend server startup complete
I1102 01:50:21.981602920    8772 grpclb_end2end_test.cc:529] starting backend server on port 14015
I1102 01:50:21.983274388    8999 server_builder.cc:269]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I1102 01:50:21.988490516    8772 grpclb_end2end_test.cc:534] backend server startup complete
I1102 01:50:21.989915575    8772 grpclb_end2end_test.cc:529] starting backend server on port 3970
I1102 01:50:21.990291853    9059 server_builder.cc:269]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I1102 01:50:21.995193283    8772 grpclb_end2end_test.cc:534] backend server startup complete
I1102 01:50:21.996930782    8772 grpclb_end2end_test.cc:529] starting balancer server on port 9749
I1102 01:50:21.997243555    9089 server_builder.cc:269]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I1102 01:50:22.003124107    8772 grpclb_end2end_test.cc:534] balancer server startup complete
I1102 01:50:22.007362220    9120 grpclb_end2end_test.cc:189] LB[0x613000009a40]: BalanceLoad
I1102 01:50:22.008509111    9120 grpclb_end2end_test.cc:193] LB[0x613000009a40]: recv msg 'initial_request {
  name: "servername_not_used"
}
'
I1102 01:50:22.008558742    9120 grpclb_end2end_test.cc:305] LB[0x613000009a40]: sleeping for 0 ms...
I1102 01:50:22.008625988    9120 grpclb_end2end_test.cc:311] LB[0x613000009a40]: Woke up! Sending response 'server_list {
  servers {
    ip_address: "\177\000\000\001"
    port: 25482
  }
  servers {
    ip_address: "\177\000\000\001"
    port: 14407
  }
  servers {
    ip_address: "\177\000\000\001"
    port: 14015
  }
  servers {
    ip_address: "\177\000\000\001"
    port: 3970
  }
}
'
I1102 01:50:22.009093491    8772 grpclb.cc:530]              Missing LB token for backend address 'ipv4:127.0.0.1:25482'. The empty token will be used instead
I1102 01:50:22.009121796    8772 grpclb.cc:530]              Missing LB token for backend address 'ipv4:127.0.0.1:14407'. The empty token will be used instead
I1102 01:50:22.009166886    8772 grpclb.cc:530]              Missing LB token for backend address 'ipv4:127.0.0.1:14015'. The empty token will be used instead
I1102 01:50:22.009185412    8772 grpclb.cc:530]              Missing LB token for backend address 'ipv4:127.0.0.1:3970'. The empty token will be used instead
I1102 01:50:22.374460325    8772 grpclb_end2end_test.cc:142] Backend: shut down
I1102 01:50:22.374519659    8772 grpclb_end2end_test.cc:553] backend about to shutdown
I1102 01:50:22.374549130    9120 grpclb_end2end_test.cc:251] LB[0x613000009a40]: done
I1102 01:50:22.375305125    8772 grpclb_end2end_test.cc:556] backend shutdown completed
I1102 01:50:22.375330640    8772 grpclb_end2end_test.cc:142] Backend: shut down
I1102 01:50:22.375345618    8772 grpclb_end2end_test.cc:553] backend about to shutdown
I1102 01:50:22.375948884    8772 grpclb_end2end_test.cc:556] backend shutdown completed
I1102 01:50:22.375971059    8772 grpclb_end2end_test.cc:142] Backend: shut down
I1102 01:50:22.375981139    8772 grpclb_end2end_test.cc:553] backend about to shutdown
I1102 01:50:22.376618743    8772 grpclb_end2end_test.cc:556] backend shutdown completed
I1102 01:50:22.376645473    8772 grpclb_end2end_test.cc:142] Backend: shut down
I1102 01:50:22.376659924    8772 grpclb_end2end_test.cc:553] backend about to shutdown
I1102 01:50:22.377225182    8772 grpclb_end2end_test.cc:556] backend shutdown completed
E1102 01:50:22.377430681    8910 round_robin.cc:280]         assertion failed: !p->shutdown



*******************************
Caught signal SIGABRT
bins/asan/grpclb_end2end_test(backtrace+0x41)[0x45bca1]
bins/asan/grpclb_end2end_test[0xfe1415]
/lib/x86_64-linux-gnu/libpthread.so.0(+0xf890)[0x7f6e52e26890]
/lib/x86_64-linux-gnu/libc.so.6(gsignal+0x37)[0x7f6e5235f067]
/lib/x86_64-linux-gnu/libc.so.6(abort+0x148)[0x7f6e52360448]
bins/asan/grpclb_end2end_test[0xa542fe]
bins/asan/grpclb_end2end_test[0x8eecc9]
bins/asan/grpclb_end2end_test[0xa34a03]
bins/asan/grpclb_end2end_test[0xa3c36b]
bins/asan/grpclb_end2end_test[0x8eecc9]
bins/asan/grpclb_end2end_test[0x8e1af4]
bins/asan/grpclb_end2end_test[0x8e09dd]
bins/asan/grpclb_end2end_test[0x6cbad4]
bins/asan/grpclb_end2end_test[0x6dccc1]
bins/asan/grpclb_end2end_test[0x757836]
bins/asan/grpclb_end2end_test[0x6db72d]
bins/asan/grpclb_end2end_test[0xa1ff92]
bins/asan/grpclb_end2end_test[0x6dd2dc]
bins/asan/grpclb_end2end_test[0x6dcc5b]
bins/asan/grpclb_end2end_test[0x711f02]
bins/asan/grpclb_end2end_test[0x7116f9]
bins/asan/grpclb_end2end_test[0x711467]
bins/asan/grpclb_end2end_test[0x82e0f5]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x8064)[0x7f6e52e1f064]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x6d)[0x7f6e5241262d]
```

**Comment by dgquintas** [comment]
Duplicate of #13040

**Comment by dgquintas** [comment]
No, this is a different one. Reopening.

**Comment by dgquintas** [comment]
FWIW, this is a known issue that should soon be fixed by #12829

**Comment by ncteisen** [comment]
#12829 is merged, fixing
